event/src/app/api/events/route.ts

```diff
@@ -0,0 +1,52 @@
+import { IUserPayload } from "@/app/api/users/login/route";
+import { NextRequest, NextResponse } from "next/server";
+import * as jose from 'jose'
+import Event, { Events } from "@/models/Event";
+import dbConnect from "@/lib/mongo";
+import { EventSchema, IEvent } from "./host/route";
+
+export async function POST(req: NextRequest) {
+
+  const body: IEvent = await req.json();
+  const validate = EventSchema.safeParse(body);
+  if (!validate.success) {
+    return NextResponse.json({
+      body: { error: 'Data Invalid', details: validate.error },
+    }, { status: 400 }
+    )
+  }
+
+  let skey: string = process.env.SECRETKEY!;
+  let cookie = req.cookies.get("accessToken")
+  const key = new TextEncoder().encode(skey)
+  if (!cookie) {
+    console.log('no cookie')
+    return NextResponse.json({
+      body: { error: 'Unauthorized', details: 'User is not authenticated. Please login' },
+    }, { status: 401 })
+  }
+
+  const { payload, protectedHeader }: { payload: IUserPayload, protectedHeader: any } = await jose.jwtVerify(cookie.value, key, {})
+  if (!payload.isSubscribed) {
+    return NextResponse.json({ error: "Unauthorized", details: "User is not subscribed to the service. Subscription is required to access this feature." }, { status: 401 })
+  }
+
+  await dbConnect();
+
+  if (!body.slug) {
+    body.slug = body.name.split(' ').join('-')
+  }
```

Comment:
Suggest to make slug required and/or auto-generate on the FE, BE can just check availability and reject if already taken.

If we want to auto-generate on BE, it'll be better to make it add a number/word at the end of the slug for duplicates, i.e. if `cool-event` is taken, auto-rename to `cool-event-2` or `cool-event-hostid` or something.

Or we can remove the slug from db and generate `slug = name.split(' ').join('-')` for the url and make sure `name` doesn't contain any `-`.

Thoughts?